internal/controllers/machinehealthcheck/machinehealthcheck_controller_test.go

```diff
@@ -1070,6 +1070,101 @@ func TestMachineHealthCheck_Reconcile(t *testing.T) {
 		}, timeout, 100*time.Millisecond).Should(Equal(1))
 	})
 
+	t.Run("when a Machine's Node has gone away without conditions", func(t *testing.T) {
+		t.Skip("skipping until made stable")
```

Comment:
Is this test flaky? (or is the skip just copy&paste?)

Reply:
they can be fixed together,

Reply:
Let's open a follow-up issue for that please

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/9903

Reply:
I don't think we should add a new skipped / flaky test
We should fix this test and remove skip (bonus points if we find the root cause and fix also the similar test we are now skipping)

Reply:
Node gone away is a bit tricky.
I followed it but can't fix it.

However i tested a new test:


```go
	t.Run("Machine's Node without conditions", func(t *testing.T) {
		g := NewWithT(t)
		cluster := createCluster(g, ns.Name)

		mhc := newMachineHealthCheck(cluster.Namespace, cluster.Name)
		mhc.Spec.UnhealthyConditions = nil

		g.Expect(env.Create(ctx, mhc)).To(Succeed())
		defer func(do ...client.Object) {
			g.Expect(env.Cleanup(ctx, do...)).To(Succeed())
		}(cluster, mhc)

		// Healthy nodes and machines.
		_, machines, cleanup1 := createMachinesWithNodes(g, cluster,
			count(3),
			firstMachineAsControlPlane(),
			createNodeRefForMachine(true),
			nodeStatus(corev1.ConditionTrue),
			machineLabels(mhc.Spec.Selector.MatchLabels),
		)
		defer cleanup1()

		//machines = append(machines)
		targetMachines := make([]string, len(machines))
		for i, m := range machines {
			targetMachines[i] = m.Name
		}
		sort.Strings(targetMachines)

		// Make sure the status matches.
		g.Eventually(func() *clusterv1.MachineHealthCheckStatus {
			err := env.Get(ctx, util.ObjectKey(mhc), mhc)
			if err != nil {
				return nil
			}
			return &mhc.Status
		}).Should(MatchMachineHealthCheckStatus(&clusterv1.MachineHealthCheckStatus{
			ExpectedMachines:    3,
			CurrentHealthy:      3,
			RemediationsAllowed: 3,
			ObservedGeneration:  1,
			Targets:             targetMachines,
			Conditions: clusterv1.Conditions{
				{
					Type:   clusterv1.RemediationAllowedCondition,
					Status: corev1.ConditionTrue,
				},
			},
		}))

		// Calculate how many Machines have health check succeeded = false.
		g.Eventually(func() (unhealthy int) {
			machines := &clusterv1.MachineList{}
			err := env.List(ctx, machines, client.MatchingLabels{
				"selector": mhc.Spec.Selector.MatchLabels["selector"],
			})
			if err != nil {
				return -1
			}

			for i := range machines.Items {
				if conditions.IsFalse(&machines.Items[i], clusterv1.MachineHealthCheckSucceededCondition) {
					unhealthy++
				}
			}
			return
		}).Should(Equal(0))

		// Calculate how many Machines have been remediated.
		g.Eventually(func() (remediated int) {
			machines := &clusterv1.MachineList{}
			err := env.List(ctx, machines, client.MatchingLabels{
				"selector": mhc.Spec.Selector.MatchLabels["selector"],
			})
			if err != nil {
				return -1
			}

			for i := range machines.Items {
				if conditions.IsTrue(&machines.Items[i], clusterv1.MachineOwnerRemediatedCondition) {
					remediated++
				}
			}
			return
		}).Should(Equal(0))
	})
```

Reply:
if you think it is okay to replace it, I will commit to this one.

Reply:
Sounds good!